### PR TITLE
[GLUTEN-2193][CH] Support function array_intersect

### PR DESCRIPTION
**Pull request by exmy**
## What changes were proposed in this pull request?

CH arrayIntersect and Spark array_intersect return the same array element but may different permutation, we temporarily ignore it.

## How was this patch tested?

UT

### CODE REVIEW

**Review by liuneng1994 — APPROVED** [review]
LGTM

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -184,6 +184,8 @@ class VeloxTestSettings extends BackendTestSettings {
     .exclude("Shuffle")
     // TODO: ArrayDistinct should handle duplicated Double.NaN
     .excludeByPrefix("SPARK-36741")
+    // TODO: ArrayIntersect should handle duplicated Double.NaN
+    .excludeByPrefix("SPARK-36754")
```
Comment (exmy) [inline gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:188]: @rui-mo Could you help verify if the handling of NaN in array-related functions in Velox is different from Spark?

Comment (rui-mo) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:188]: @Ma-Jian1 Could you help check? I remember you have some findings here.

Comment (exmy) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:188]: Can this pr be merged?

Comment (rui-mo) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:188]: Yes, let's merge this PR first.

**Review by rui-mo — APPROVED** [review]

**Pull request merged**